webapp/src/components/modals/configModal.tsx

```diff
@@ -0,0 +1,760 @@
+import React, {useEffect, useState} from 'react';
+
+import Button from 'react-bootstrap/Button';
+import Modal from 'react-bootstrap/Modal';
+import Form from 'react-bootstrap/Form';
+import Table from 'react-bootstrap/Table';
+import ButtonGroup from 'react-bootstrap/ButtonGroup';
+import {OverlayTrigger, Tooltip, ToggleButton} from 'react-bootstrap';
+
+import Select, {MultiValue, SingleValue} from 'react-select';
+
+import {useSelector} from 'react-redux';
+
+import {GlobalState} from 'mattermost-redux/types/store';
+
+import {fetchChannels, fetchTeams} from 'api/api_wrapper';
+
+import {DeleteSvg, EditSvg} from '../svgIcons/svg';
+
+import './styles.scss';
+
+type Props = {
+    visibility: boolean;
+    setVisibility: React.Dispatch<React.SetStateAction<boolean>>;
+    configIndex: number | null;
+    config: Configs[];
+    onChange: (config: Configs[]) => void;
+    modalHeader: string;
+}
+
+const ConfigModal = ({visibility, setVisibility, configIndex, config, onChange, modalHeader}: Props) => {
+    const guest = [
+        {name: 'true', value: 'true'},
+        {name: 'false', value: 'false'},
+    ];
+    const actionTypes = [
+        {name: 'button', value: 'button'},
+        {name: 'automatic', value: 'automatic'},
+    ];
+
+    const actionElement: Actions = {
+        actionType: '',
+        actionName: '',
+        actionDisplayName: '',
+        channelsAddedTo: [''],
+        actionSuccessfullMessage: [''],
+    };
+
+    const newAction: Actions[] = [];
+
+    const newConfig: Configs = {
+        teamName: '',
+        delayInSeconds: 0,
+        message: [''],
+        includeGuests: '',
+        attachmentMessage: [''],
+        actions: newAction,
+    };
+
+    const [show, setShow] = useState(true);
+    const [isConfigVisible, setIsConfigVisible] = useState(true);
+    const [isActionVisible, setIsActionVisible] = useState(false);
+    const [isDeleteVisible, setIsDeleteVisible] = useState(false);
+
+    const [existingConfig, setExistingConfig] = useState(configIndex === null ? newConfig : config[configIndex]);
+
+    const [teamName, setTeamName] = useState(existingConfig.teamName);
+    const [delay, setDelay] = useState(existingConfig.delayInSeconds);
+    const [message, setMessage] = useState(existingConfig.message);
+    const [attachmentMessage, setAttachmentMessage] = useState(existingConfig.attachmentMessage ?? ['']);
```

Comment:
```suggestion
    const [attachmentMessages, setAttachmentMessages] = useState(existingConfig.attachmentMessage ?? ['']);
```
If this is an array of string. I think there with be many messages and not just one. Right?

Reply:
For this release, we are treating array of string as one string only

Reply:
What does it mean?

Reply:
The current form available to add/edit configs only accepts single string as an attachment message but the backend still wants the attachment message to be an array of string